Make OCF Perceiver_Pytorch pip installable

jacobbieker opened the pull request:
Adds OCF's pytorch to PyPi with name `perceiverio` so that we can then use `pip install perceiverio` the same as Skillful Nowcasting and MetNet models.

peterdudfield commented on setup.py:
```diff
@@ -1,17 +1,21 @@
 
 setup(
-    name="perceiver-pytorch",
+    name="perceiverio",
```
You happy with the name?

jacobbieker replied:
I'm not sure its the best name, but `perceiver` is taken, same with a lot of other shortish names. But totally open to any other suggestions!

peterdudfield replied:
yea, maybe perceiverio is ok. It has some none perceiver io stuff in it, so i dont know how to reflect that. Maybe perceiverio is good, and I'm over thinking it

JackKelly replied:
Yeah, `perceiverio` is OK. 

`perceiver-io` is perhaps more readable, and Pip is happy with hyphens.

Something like `perceivermodel` or `perceiver-model` might be worth considering:  It's more obviously two words (instead of 
`perceiverio` which is all-too-tempting to read as a single word: 'perceivereo') and it is perhaps more general than `perceiverio`, so it consistent with the fact that the code includes both `perceiver` and `perceiver IO` code.  But maybe I'm over-thinking this :)

jacobbieker replied:
Okay, yeah, I'll change it to `perceiver-model`, sounds like a good way to make it sound more general!

peterdudfield commented:
Is it worth a little tidy up of the following files

from perceiver_pytorch.perceiver_pytorch import Perceiver
from perceiver_pytorch.perceiver_io import PerceiverIO, PerceiverLM
from perceiver_pytorch.multi_perceiver_pytorch import MultiPerceiver

Some of them have duplicate functions in them.

Also would love a bit of descirption for each one, maybe + paper reference. 
I think Perceiver was made by the original repo - so maybe just concentrate on description of the new models

jacobbieker commented:
> Is it worth a little tidy up of the following files
> 
> from perceiver_pytorch.perceiver_pytorch import Perceiver
> from perceiver_pytorch.perceiver_io import PerceiverIO, PerceiverLM
> from perceiver_pytorch.multi_perceiver_pytorch import MultiPerceiver
> 
> Some of them have duplicate functions in them.
> 
> Also would love a bit of descirption for each one, maybe + paper reference.
> I think Perceiver was made by the original repo - so maybe just concentrate on description of the new models

Okay, yeah, I'll make changes to those

jacobbieker commented:
@peterdudfield I've added a bit more documentation and paper links, and refactored some more of the common functions. I also removed the experimental and PerceiverLM as I don't think either of those are useful to what we are doing.

peterdudfield reviewed (approved):
Looks all good, 
You may want to get @JackKelly to have a look, but looks pretty good. 
let me know when you have merged and I can try pip installing it

JackKelly reviewed (approved):
LGTM!  Thanks for refactoring the code!  Great work :)

The only other thing I can think to do is to remove all mention on `experimental` from the README
